### ISSUE TITLE
Module not found on running the task

### ISSUE THREAD

**Issue by zizizi17** [ui-jar/ui-jar#23]
Hello. 
I got the errors while running start-ui-jar-example script.
_ERROR in ./node_modules/ui-jar/dist/src/app/app.module.js
Module not found: Error: Can't resolve '../../../temp/__ui-jar-temp' in 'E:\work\portal\node_modules\ui-jar\dist\src\app'_
![1](https://user-images.githubusercontent.com/10571073/38083127-6ad08922-3351-11e8-8d0c-64a46c402a90.PNG)
Can you recommend what i do wrong?

**Comment by nording** [comment]
Is that the complete stacktrace? Does it say something about "Generated resources completed" before?

**Comment by zizizi17** [comment]
Yes, it says such

> Generating resources...
Cannot read property 'extendClasses' of undefined
** NG Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **

And then info about chunks and ERROR

**Comment by nording** [comment]
Allright, what version of UI-jar is installed? What version of Angular CLI are you using?

**Comment by zizizi17** [comment]
angular-cli is 1.6.1 version, ui-jar: 1.0.0

**Comment by nording** [comment]
Okey, does the error also occurs with UI-jar version 1.1.0?

**Comment by zizizi17** [comment]
Yeap, the same issue

**Comment by nording** [comment]
Are you using the Angular CLI example right out of the box or have you made any changes? If, what changes? This will help me to repro this.

**Comment by zizizi17** [comment]
I haven't done any changes. I just copied the ui-jar folder and scripts from package.json

**Comment by zizizi17** [comment]
Where the error with Cannot read property "extendClass" of undefined be caused?

**Comment by nording** [comment]
Have you added any UI-jar annotation to your components you would like go view and the test to them? (The reason the error occurs may be that)

**Comment by zizizi17** [comment]
Can you add here the example?
I added the @component & @group annotation to the component, but I add no tests for component.

**Comment by nording** [comment]
Okey check out https://github.com/ui-jar/ui-jar/blob/master/README.md#example-usage-basic and the test code there.

**Comment by zizizi17** [comment]
But how can it be?
I removed all the annotation from the example, Button component, from test and it runs properly. And in my case I cannot event start the app

**Comment by nording** [comment]
Not sure, i'm trying to repro this but I have not found the case yet. Is the source code to your project closed or is it available somewhere? I would like to find a case where I can repro this.

**Comment by zizizi17** [comment]
It is closed repo((

**Comment by nording** [comment]
I found a way to reproduce it! :) will fix it!

**Comment by zizizi17** [comment]
Can i fix it by myself for my project? 
If no, when there will be update?

**Comment by nording** [comment]
I could reproduce it when I created a interface that extend another interface. If you wan't you could download the master, build it and try it from there. Otherwise I will try to release a new version later this week.

**Comment by zizizi17** [comment]
It still does not work((

**Comment by nording** [comment]
Do you get same error or other?

**Comment by zizizi17** [comment]
A bit another 
![2](https://user-images.githubusercontent.com/10571073/38094469-7b5f79a4-3376-11e8-9da5-dac28b1cd756.PNG)

**Comment by nording** [comment]
Allright, I think I got that error message with a version of Angular CLI, worked when I updated Angular CLI. Could you try that? This thread Night help you https://github.com/angular/angular-cli/issues/8036

**Comment by nording** [comment]
Have you added UI-jar as an app to the config in your angular-cli.json?
See example app config herr: 
https://github.com/ui-jar/ui-jar/blob/master/examples/example-app-with-angular-cli/.angular-cli.json
Does it work then?

**Comment by zizizi17** [comment]
Do you mean the downgrading of the @angular/cli?

**Comment by nording** [comment]
There was a few different tips there, not sure if they work (look like the "Field 'browser' doesn't contain a valid alias configuration" is a bit hard to derive).

I have been working on trying to repro the last error you get,
Step 1: I created a new Angular CLI project (Angular CLI 1.7.3)
Step 2: Copied UI-jar folder from the angular-cli-example
Step 3: Copied run config in package.json "ui-jar & start-ui-jar-example".
Step: 4: After that I copied "UI-jar" app config in angular-cli.json from the angular-cli-example.
Step 5: Then I ran "npm run start-ui-jar-example", it started as expected.

I guess you have tried this steps already? If not, could you try it?

This is the package.json in my test project:
```
{
  "name": "test-project",
  "version": "0.0.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "build": "ng build --prod",
    "test": "ng test",
    "lint": "ng lint",
    "e2e": "ng e2e",
    "ui-jar": "node node_modules/ui-jar/dist/bin/cli.js --directory ./src/app/ --includes \\.ts$",
    "start-ui-jar-example": "npm run ui-jar && ng serve --app ui-jar"
  },
  "private": true,
  "dependencies": {
    "@angular/animations": "^5.2.0",
    "@angular/common": "^5.2.0",
    "@angular/compiler": "^5.2.0",
    "@angular/core": "^5.2.0",
    "@angular/forms": "^5.2.0",
    "@angular/http": "^5.2.0",
    "@angular/platform-browser": "^5.2.0",
    "@angular/platform-browser-dynamic": "^5.2.0",
    "@angular/router": "^5.2.0",
    "core-js": "^2.4.1",
    "rxjs": "^5.5.6",
    "ui-jar": "^1.1.0",
    "zone.js": "^0.8.19"
  },
  "devDependencies": {
    "@angular/cli": "~1.7.3",
    "@angular/compiler-cli": "^5.2.0",
    "@angular/language-service": "^5.2.0",
    "@types/jasmine": "~2.8.3",
    "@types/jasminewd2": "~2.0.2",
    "@types/node": "~6.0.60",
    "codelyzer": "^4.0.1",
    "jasmine-core": "~2.8.0",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~2.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-coverage-istanbul-reporter": "^1.2.1",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.2",
    "ts-node": "~4.1.0",
    "tslint": "~5.9.1",
    "typescript": "~2.5.3"
  }
}
```

**Comment by nording** [comment]
I did try one more thing:
Copied the "src/app/ui" folder with example components from angular-cli-example to test-project.
Then when I started it, I got the following errors:

```
ERROR in ./src/app/ui/buttons/button-primary.directive.ts
Module build failed: Error: path\test-project\src\app\ui\buttons\button-primary.directive.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include' property.
    at AngularCompilerPlugin.getCompiledFile (path\test-project\node_modules\@ngtools\webpack\src\angular_compiler_plugin.js:674:23)
    at plugin.done.then (path\test-project\node_modules\@ngtools\webpack\src\loader.js:467:39)
    at <anonymous>
```
I could resolve that by adding following to file "ui-jar/tsconfig.app.json":
```
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/app",
    "baseUrl": "./",
    "module": "es2015",
    "types": []
  },
  "exclude": [
    "test.ts",
    "**/*.spec.ts",
    "../src/**/*.spec.ts" <-- added
  ],
  "include": [
    "**/*.ts", <-- added
    "../src/**/*.ts" <-- added
  ]
}
```

Have you tried that?

**Comment by zizizi17** [comment]
Anyway it has the error with the 'browser'

**Comment by nording** [comment]
Have you tried a clean setup with Angular CLI? Do you have any problem of you create a new project as i describe above and just copied the examples? (Trying till find out when the problem occurs)

**Comment by zizizi17** [comment]
I don't know how, but it started to work

**Comment by nording** [comment]
Great! Let me know if you find out what you did :)

**Comment by zizizi17** [comment]
As I can judge, I did not add the environments folder in the ui-jar.

**Comment by nording** [comment]
Allright thanks! It's great to know if someone else get same problem.

**Comment by zizizi17** [comment]
Okay, one more question. 
I launch the ui-jar and got the error about "cannot find module ...". And it's all the imports in all the files across the app. 
I think that it can be related to the baseUrl, maybe, you got the same error?

**Comment by nording** [comment]
Do you get that error message in your browser console?
Are we talking about the baseUrl in tsconfig.app.json?
Have you change it to any specific other than "./" ?

**Comment by zizizi17** [comment]
I got error messages in the console.
Yes, we talk about tsconfig.app.json in the ui-jar folder.
I changed it to the "../", because I have the absolute path in the imports through application.

**Comment by nording** [comment]
Allright, I will take a look at that regarding absolute paths! At the moment all paths is resolved relative to file location, but it should respect if other is specified in tsconfig.

**Comment by nording** [comment]
I have tried to repro your problem with absolute paths,
The repro project has the following structure:
```
/src
 /app
  /abc
   abc.component.ts
   abc.service.ts
     ...
/ui-jar
 tsconfig.app.json
...
```
I added one absolute import path to src/app/abc/abc.component.ts:
```
import { AbcService } from 'app/abc/abc.service';
```
My ui-jar/tsconfig.app.json file:
```
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/app",
    "baseUrl": "../", <-- notice that I changed my baseUrl path
    "module": "es2015",
    "types": [],
    "paths": {
      "app/*": ["app/*"] <-- I needed to have paths settings for my absolute imports
    }
  },
  "exclude": [
    "test.ts",
    "**/*.spec.ts",
    "../src/**/*.spec.ts"
  ],
  "include": [
    "**/*.ts",
    "../src/**/*.ts"
  ]
}
```
Then when I started with "npm run start-ui-jar-example" I got the following error message in console:
```
ERROR in src/app/abc/abc.component.ts(3,40): error TS2307: Cannot find module 'app/abc/abc.service'.
```

I could fix it by doing this:
```
{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "outDir": "../out-tsc/app",
    "baseUrl": "../src", <-- Notice that I added "src" here, instead of just "../"
    "module": "es2015",
    "types": [],
    "paths": {
      "app/*": ["app/*"]
    }
  },
  "exclude": [
    "test.ts",
    "**/*.spec.ts",
    "../src/**/*.spec.ts"
  ],
  "include": [
    "**/*.ts",
    "../src/**/*.ts"
  ]
}
```
Then when I restarted "npm run start-ui-jar-example", it all compiled, and worked as expected.
@zizizi17 could you verify your config like I have done? I think of "baseUrl" property and that your paths matches with it like in the example above.

**Comment by zizizi17** [comment]
Yes, it works perfectly now.

Can i contact with you in some social network, I have a few questions more. It would be much quicker to communicate?

**Comment by nording** [comment]
Great! My Twitter is @knording - send a message over there.

**Comment by nording** [comment]
I'm closing this, I released v1.1.1 today, it is available on npm!
